migration/data.rst

```diff
@@ -0,0 +1,73 @@
+.. _data_migration:
+
+Data Migration
+==============
+
+Migrating from one database to another has traditionally been difficult. There are two traditional approaches: using a dump and restore that takes downtime, or updating application logic to write to two databases and then switch to the new one.
+
+Now there's an easier way. PostgreSQL 10 added logical replication to stream data to a new database even while the existing database continues to sustain application load. It's as if the application automatically writes to two databases rather than one, except with perfect transactional logic.
```

Comment:
People might be confused and think that we require Postgres 10 on the source - can we rewrite this to be less focused on the Postgres 10 logical replication?

For context, Citus Warp uses logical decoding, which is part of Postgres starting 9.4.